compiler/test/stdlib/result.test.gr

```diff
@@ -0,0 +1,99 @@
+import Result from 'result';
+
+# isOk
+assert Result.isOk(Ok(1)) == true
+assert Result.isOk(Err("Ooops")) == false
+
+# isErr
+assert Result.isErr(Ok(1)) == false
+assert Result.isErr(Err("Ooops")) == true
+
+# toOption
+assert Result.toOption(Ok(1)) == Some(1)
+assert Result.toOption(Err("Ooops")) == None
+
+# mapOk
+assert Result.mapOk(Ok(1), a => "some other type") == Ok("some other type")
+assert Result.mapOk(Err("Ooops"), incr) == Err("Ooops")
```

Comment:
```suggestion
assert Result.mapOk(Err("Ooops"), (v) => fail 'Shouldn\'t run') == Err("Ooops")
```

Reply:
I think we should make sure to use failures on anything that shouldn't be called to make sure some refactoring isn't done that hoists the function to always be called then throws away the result.

Reply:
Can also use double quoted strings for these:
```suggestion
assert Result.mapOk(Err("Ooops"), (v) => fail "Shouldn't run") == Err("Ooops")
```